crates/cxx-qt-lib/src/core/qobject.rs

```diff
@@ -64,12 +67,10 @@ pub trait QObjectExt {
     fn object_name(&self) -> QString;
 
     /// Returns a mutable pointer to the parent object.
-    fn parent(&self) -> *mut Self;
+    fn parent(&self) -> *mut QObjectExternal;
```

Comment:
Wonder if this should be a QObject not external? As i might want to ask for the parent and then downcast from QObject* to a T* ? Guess we'd need to cast the QObjectExternal -> QObject in the parent method?

Or do we think that we do not need that and QObjectExternal is fine? :thinking: